### PR TITLE
Added QOI reading

### LINKED ISSUE #6844: QOI support
### What did you do?

Tried to open a QOI image ( https://qoiformat.org/ ) with Pillow (`Image.open("picture.qoi")`)

The picture is in this zip file (a 2-seconds stick-man) : [picture.zip](https://github.com/python-pillow/Pillow/files/10327604/picture.zip)

### What did you expect to happen?

I would love Pillow to recognize the file type and open the image.

### What actually happened?

Pillow can't recognize the QOI format.

```python
>>> from PIL import Image
>>> Image.open("picture.qoi")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python3.10/site-packages/PIL/Image.py", line 3186, in open
    raise UnidentifiedImageError(
PIL.UnidentifiedImageError: cannot identify image file 'picture.qoi'
```

### What are your OS, Python and Pillow versions?

* OS: Manjaro Linux
* Python: 3.10.8
* Pillow: 9.3.0

### To replicate

```python
from PIL import Image
Image.open("picture.qoi")
```

### PR DESCRIPTION
**Pull request by radarhere**
Resolves #6844

Adds support for reading the QOI image format - https://qoiformat.org/.

The specification is https://qoiformat.org/qoi-specification.pdf
I created the test images using https://www.aconvert.com/image/png-to-qoi/

### CODE REVIEW

File: docs/handbook/image-file-formats.rst [hunk 1]
```diff
@@ -1544,6 +1544,13 @@ The :py:meth:`~PIL.Image.Image.save` method can take the following keyword argum
 
     .. versionadded:: 5.3.0
 
+QOI
+^^^
+
+.. versionadded:: 9.5.0
+
+Pillow identifies and reads QOI images.
```
Comment (hugovk) [inline docs/handbook/image-file-formats.rst:1552]:
```suggestion
Pillow identifies and reads images in Quite OK Image format.
```

File: Tests/test_file_qoi.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import pytest
+
+from PIL import Image, QoiImagePlugin
+
+from .helper import assert_image_equal_tofile, assert_image_similar_tofile
+
+
+class TestFileQOI:
```
Comment (hugovk) [inline Tests/test_file_qoi.py:8]: We could maybe skip the class here and use plain, top-level functions?

File: docs/releasenotes/9.5.0.rst [hunk 3]
```diff
@@ -28,6 +28,11 @@ TODO
 API Additions
 =============
 
+QOI file format
+^^^^^^^^^^^^^^^
+
+Pillow can now read QOI images.
```
Comment (hugovk) [inline docs/releasenotes/9.5.0.rst:34]:
```suggestion
Pillow can now read images in Quite OK Image format.
```

File: src/PIL/QoiImagePlugin.py [hunk 4]
```diff
@@ -0,0 +1,106 @@
+#
+# The Python Imaging Library.
+# $Id$
```
Comment (hugovk) [inline src/PIL/QoiImagePlugin.py:3]: We could skip this for new files:
```suggestion
```

File: Tests/test_file_qoi.py (hunk 2 above)
Comment (radarhere) [reply Tests/test_file_qoi.py]: You're right, I've added a commit for this.

File: src/PIL/QoiImagePlugin.py (hunk 4 above)
Comment (radarhere) [reply src/PIL/QoiImagePlugin.py:3]: Do you have any ideas what the original meaning was? Is this some sort of placeholder?

Comment (hugovk) [reply src/PIL/QoiImagePlugin.py:3]: It's old VCS stuff, they'd replace that when checking out.

https://stackoverflow.com/q/2763070/724176

Comment (radarhere) [reply src/PIL/QoiImagePlugin.py]: Thanks, I've removed it.

**Review by hugovk — APPROVED** [review]

File: src/PIL/QoiImagePlugin.py (hunk 4 above)
Comment (radarhere) [reply src/PIL/QoiImagePlugin.py]: Is there any downside to removing it for all existing files as well? https://github.com/python-pillow/Pillow/pull/7009

Comment (hugovk) [reply src/PIL/QoiImagePlugin.py]: Just the downsides of a big chunk of code churn. Perhaps we remove it incrementally as we touch the files for other reasons?

Comment (radarhere) [reply src/PIL/QoiImagePlugin.py]: Ok, thanks. I've closed the PR.

**Pull request merged**